tests/utilities/test_distributed.py

```diff
@@ -53,3 +59,25 @@ def foo():
 
         x = foo()
         assert x is None
+
+
+def _test_collect_states(rank, worldsize):
+    os.environ["MASTER_ADDR"] = "localhost"
+
+    # initialize the process group
+    torch.distributed.init_process_group("nccl", rank=rank, world_size=worldsize)
+
+    state = {"something": torch.tensor([rank])}
+    collected_state = _collect_states_on_rank_zero(state, device=torch.device(f"cuda:{rank}"))
+    if rank == 1:
+        assert collected_state is None
+    else:
+        print(collected_state)
+        assert collected_state == {1: {"something": torch.tensor([1])}, 0: {"something": torch.tensor([0])}}
+
+
+@RunIf(skip_windows=True, min_gpus=2, min_torch="1.10")
```

Comment:
why is 1.10 required?

Reply:
It didn't seem to be working with previous version. Diving into this right now, but having some issues.

Reply:
Yes, it seems to be failing.